pygeoapi/provider/xarray_.py

```diff
@@ -67,8 +68,19 @@ def __init__(self, provider_def):
                     open_func = xarray.open_mfdataset
                 else:
                     open_func = xarray.open_dataset
+            if provider_def['data'].startswith('s3://'):
```

Comment:
Given the `options` proposal in this PR, then would something like the below help?

```python
            if provider_def['data'].startswith('s3://'):
                LOGGER.debug('Data is stored in S3 bucket.')
                if 's3' in provider_def.get('options', {}):
                    s3_options = provider_def['options']['s3']
                else:
                    s3_options = {}

                data_to_open = fsspec.get_mapper(self.data, s3_options)
                LOGGER.debug('Completed S3 open function')
            else:
                data_to_open = self.data
```